internal/services/ticker/schedule_workflow.go

```diff
@@ -91,24 +88,28 @@ func (t *TickerImpl) handleScheduleWorkflow(ctx context.Context, scheduledWorkfl
 		return nil
 	}
 
-	// schedule the workflow
-	_, err = s.NewJob(
-		gocron.OneTimeJob(
-			gocron.OneTimeJobStartDateTime(triggerAt),
-		),
-		gocron.NewTask(
-			t.runScheduledWorkflow(tenantId, workflowVersionId, scheduledWorkflowId, scheduledWorkflow),
-		),
-	)
+	// create a cancellation context for this specific scheduled workflow
+	cancelCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Why don't we set a timeout on this context? We know what the lookahead time should be; we can probably set the timeout as twice the lookup time pretty safely

Reply:
pushed